lib/rex/post/meterpreter/ui/console.rb

```diff
@@ -100,17 +100,21 @@ def queue_cmd(cmd)
   def run_command(dispatcher, method, arguments)
     begin
       super
-    rescue Timeout::Error
-      log_error("Operation timed out.")
-    rescue RequestError => info
-      log_error(info.to_s)
-    rescue Rex::InvalidDestination => e
-      log_error(e.message)
-    rescue ::Errno::EPIPE, ::OpenSSL::SSL::SSLError, ::IOError
-      self.client.kill
-    rescue  ::Exception => e
-      log_error("Error running command #{method}: #{e.class} #{e}")
-      elog(e)
+    rescue Exception => e
+      unless self.client.on_error_proc && self.client.on_error_proc.call(e) == :handled
```

Comment:
This might read better as a guard clause:

```ruby
is_error_handled = self.client.on_error_proc && self.client.on_error_proc.call(e) == :handled
return if is_error_handled

case e
  ...
end
```